From 1afe878ce77c1208e66501b0a05ae8ae3a94d1ac Mon Sep 17 00:00:00 2001
From: Stefano Rivera <stefano@rivera.za.net>
Date: Wed, 9 Apr 2025 19:57:08 -0400
Subject: [PATCH] Manpage formatting tweaks

Following on from GH-132291 this is the second part of the patch from
https://bugs.debian.org/1101406

This tweaks the formatting of a few bits of the manpage.

Forwarded: https://github.com/python/cpython/pull/132338
Bug-Debian: https://bugs.debian.org/1101406
---
 Misc/python.man | 36 +++++++++++++++++++-----------------
 1 file changed, 19 insertions(+), 17 deletions(-)

--- a/Misc/python.man
+++ b/Misc/python.man
@@ -162,7 +162,7 @@ compilation options).
 Ignore environment variables like PYTHONPATH and PYTHONHOME that modify
 the behavior of the interpreter.
 .TP
-.B \-h ", " \-? ", "\-\-help
+.BR \-h ", " \-? ", " \-\-help
 Prints the usage for the interpreter executable and exits.
 .TP
 .B "\-\-help\-env"
@@ -171,7 +171,6 @@ Prints help about Python-specific enviro
 .B "\-\-help\-xoptions"
 Prints help about implementation-specific \fB\-X\fP options and exits.
 .TP
-.TP
 .B "\-\-help\-all"
 Prints complete usage information and exits.
 .TP
@@ -243,7 +242,7 @@ twice, print a message for each file tha
 searching for a module.  Also provides information on module cleanup
 at exit.
 .TP
-.B \-V ", " \-\-version
+.BR \-V ", " \-\-version
 Prints the Python version number of the executable and exits.  When given
 twice, print more information about the build.
 
@@ -256,22 +255,24 @@ to
 The simplest settings apply a particular action unconditionally to all warnings
 emitted by a process (even those that are otherwise ignored by default):
 
-  -Wdefault  # Warn once per call location
-  -Werror    # Convert to exceptions
-  -Walways   # Warn every time
-  -Wall      # Same as -Walways
-  -Wmodule   # Warn once per calling module
-  -Wonce     # Warn once per Python process
-  -Wignore   # Never warn
+.nf
+  \-Wdefault  # Warn once per call location
+  \-Werror    # Convert to exceptions
+  \-Walways   # Warn every time
+  \-Wall      # Same as \-Walways
+  \-Wmodule   # Warn once per calling module
+  \-Wonce     # Warn once per Python process
+  \-Wignore   # Never warn
+.fi
 
 The action names can be abbreviated as desired and the interpreter will resolve
 them to the appropriate action name. For example,
 .B \-Wi
 is the same as
-.B \-Wignore .
+.BR \-Wignore .
 
 The full form of argument is:
-.IB action:message:category:module:lineno
+.IB action : message : category : module : lineno
 
 Empty fields match all values; trailing empty fields may be omitted. For
 example
@@ -457,7 +458,7 @@ is an empty string; if
 is used,
 .I sys.argv[0]
 contains the string
-.I '\-c'.
+.RI ' \-c "' ."
 Note that options interpreted by the Python interpreter itself
 are not placed in
 .IR sys.argv .
@@ -584,8 +585,8 @@ purpose is to allow repeatable hashing,
 interpreter itself, or to allow a cluster of python processes to share hash
 values.
 
-The integer must be a decimal number in the range [0,4294967295].  Specifying
-the value 0 will disable hash randomization.
+The integer must be a decimal number in the range [0,4\|294\|967\|295].
+Specifying the value 0 will disable hash randomization.
 .IP PYTHONHOME
 Change the location of the standard Python libraries.  By default, the
 libraries are searched in ${prefix}/lib/python<version> and
@@ -607,7 +608,7 @@ This is equivalent to the \fB\-X int_max
 .IP PYTHONIOENCODING
 If this is set before running the interpreter, it overrides the encoding used
 for stdin/stdout/stderr, in the syntax
-.IB encodingname ":" errorhandler
+.IB encodingname : errorhandler
 The
 .IB errorhandler
 part is optional and has the same meaning as in str.encode. For stderr, the
